src/components/Grants/GrantCard/ProgressBarTooltip.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+
+import useFormatMessage, { useIntl } from 'decentraland-gatsby/dist/hooks/useFormatMessage'
+import Time from 'decentraland-gatsby/dist/utils/date/Time'
+import { Popup } from 'decentraland-ui/dist/components/Popup/Popup'
+
+import { GrantAttributes } from '../../../entities/Proposal/types'
+import { CLIFF_PERIOD_IN_DAYS } from '../../../entities/Proposal/utils'
+import { formatDate } from '../../../modules/time'
+
+interface Props {
+  grant: GrantAttributes
+  isInCliff: boolean
+  children: React.ReactNode
+}
+
+function ProgressBarTooltip({ grant, isInCliff, children }: Props) {
+  const t = useFormatMessage()
+  const intl = useIntl()
+
+  const { contract, tx_amount, token, enacted_at } = grant
+  const isOTP = !contract
+  const vestedAmount = (contract ? contract.vestedAmount : tx_amount) || 0
+  const releasedAmount = !isOTP ? contract.released : 0
+
+  let textToShow = ''
```

Comment:
This can be in a helper function to make this component easier to read
`const text = getText(...)`
But maybe it will have too many arguments...

Reply:
yep, I don't think it's worth it. It has too many arguments...